packages/javascript/bh-shared-ui/src/components/PasswordDialog/PasswordDialog.tsx

```diff
@@ -77,13 +80,17 @@ const PasswordDialog: React.FC<{
         }
     }, [open, reset, initialNeedsPasswordReset, setValue]);
 
-    const handleOnSave = (data: { password: string; confirmPassword: string; needsPasswordReset: boolean }) => {
-        return onSave({
-            userId: userId,
-            secret: data.password,
-            needsPasswordReset: Boolean(data.needsPasswordReset),
-        });
-    };
+    const handleOnSave = useCallback(
+        (data: ChangePasswordFormInputs) => {
```

Comment:
there seems to be a mixing of the words "secret" and "password" throughout this component. if there are two different names for the same thing, can we pick one and make it consistent

Reply:
We discussed offline, but it seems API related uses `secret` while UI related uses `password`. I'm open to refactoring but it might be out of scope for this particular task and I'd like to open the discussion to the wider engineering team / verify it's something we want to do.